pkg/resourcemanager/webhook/highavailabilityconfig/handler.go

```diff
@@ -150,7 +156,7 @@ func (h *Handler) handleDeployment(
 	}
 
 	h.mutateNodeAffinity(
-		failureToleranceType,
+		failureToleranceType != nil || isZonePinningEnabled,
```

Comment:
Can you add a `TODO` comment that we remove the `failureToleranceType != nil` condition in the future?